### DISCUSSION TITLE
Date Picker Help

### DISCUSSION THREAD

**Discussion by atlys-hein** [heroui-inc/heroui#3860] [Help]
Can anyone help me with the DatePicker please.

I am using the showMonthAndYearPickers on the DatePicker and what I am trying to achieve is that the Picker Wrapper closes automatically when the year and the month have been selected instead of have to click on the button again at the top the close the Month and Year picker. 

I have tried changing the calendar props classnames on the DatePicker and setting the opacity of the pickerWrapper and a few other things with no luck.

If anyone has any idea of how to achieve this it would be awesome. 
Thank you in advance 

This is my date picker without any of the changes for context
```tsx
 <DatePicker
            onChange={(value) =>
              handleInputChange("dateOfBirth", value.toString())
            }
            label="Birth date"
            className={`col-span-1`}
            isRequired
            size="md"
            showMonthAndYearPickers
            labelPlacement="outside"
            variant="bordered"
            value={
              isPlainDateString(localTraveler.dateOfBirth)
                ? parseDate(localTraveler.dateOfBirth)
                : undefined
            }
            maxValue={
              traveler.travelerType === TravelerType.PRIMARY
                ? parseDate(
                    new Date().toISOString().split("T")[0] as string
                  )?.subtract({ years: 18 })
                : undefined
            }
            isInvalid={
              validation.dateOfBirth?.isValidated &&
              !!validation.dateOfBirth?.error
            }
            errorMessage={validation.dateOfBirth?.error || undefined}
          />
```

**Comment by CanRau** [comment]
Would like this as well I think as I wasn't sure what to do at first when the month & year picker didn't close on selection 😅 
Though I see how this could also potentially be confusing 🤔

**Comment by darknight92ss** [comment]
its sad there has been no advancement on such a vital feature of the calander component
